src/Symfony/Component/Message/Asynchronous/Middleware/SendMessageToProducersMiddleware.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Message\Asynchronous\Middleware;
+
+use Symfony\Component\Message\Asynchronous\ConsumedMessage;
+use Symfony\Component\Message\Asynchronous\Routing\ProducerForMessageResolverInterface;
+use Symfony\Component\Message\MessageBusMiddlewareInterface;
+
+/**
+ * @author Samuel Roze <samuel.roze@gmail.com>
+ */
+class SendMessageToProducersMiddleware implements MessageBusMiddlewareInterface
+{
+    /**
+     * @var ProducerForMessageResolverInterface
+     */
+    private $producerForMessageResolver;
+
+    public function __construct(ProducerForMessageResolverInterface $producerForMessageResolver)
+    {
+        $this->producerForMessageResolver = $producerForMessageResolver;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle($message, callable $next)
+    {
+        if ($message instanceof ConsumedMessage) {
+            $message = $message->getMessage();
+        } elseif (!empty($producers = $this->producerForMessageResolver->getProducersForMessage($message))) {
```

Comment:
`getProducersForMessage()` => `getSender()`

Reply:
As I guess we'll rename `Producer` to `Sender` I see where you are going. But you replace plural by singular here and that's an interesting place for me to explain why it's plural.

At the beginning it was returning `ProducerInterface|null` and I had a collection of producer that was dispatching to a set of producers to handle this plural. But there's one use-case that didn't work:
 
> When I dispatch a message "Foo" in the message bus
> Then I want the message to be sent to the producer "FooProducer"
> And I want the message to be handled by the handler

i.e. I want to send the message to _something_ *and* I want to call the local handler. This can be useful for audit purposes, duplication, etc... To make this use case happening,  [you just need to have to use a `null` handler](https://github.com/symfony/symfony/pull/24411/files/1202153a9bf3408f1ca8501cb201f22c8d0bc013#diff-81bf973d4d1476edf862acd859850efaR45). (See the [note in `README.md](https://github.com/symfony/symfony/pull/24411/files/1202153a9bf3408f1ca8501cb201f22c8d0bc013#diff-aa7b060ee5794d3e1e6dd4b34a1c996fR172))

Therefore, I'd say `getProducersForMessage()` => `getSenders()`